packages/wdio-config/src/node/ConfigParser.ts

```diff
@@ -305,6 +312,18 @@ export default class ConfigParser {
             throw new Error('The --multi-run flag requires that either the --spec or --suite flag is also set')
         }
 
+        // remove glob-pattern from CLI spec arg before filtering
+        if (this._config.spec !== undefined && this._config.spec.length > 0 && this.isGlobPattern(this._config.spec[0])){
+            this._config.spec[0] = this.removeGlobPatternFromSpecCliArg(this._config.spec[0])
+        }
+
+        // filter specs by keyword
+        if (this._config.spec !== undefined){
+            if (!this.cliSpecArgContainsPathstoSpecs(this._config.spec!)){
+                specs = this.filterSpecsFromConfigFileByKeyword(this._config.spec[0], specs)
+            }
+        }
```

Comment:
Never do nested `if`s , especially when there are not further `else`s. It seems to me that we use `cliSpecArgContainsPathstoSpecs` only once so no need to put this into an extra method, instead:
```suggestion
        // filter specs by keyword
        if (this._config.spec){
            specs = this.filterSpecsFromConfigFileByKeyword(this._config.spec[0], specs)
        }
```

`filterSpecsFromConfigFileByKeyword` should handle cases where `this._config.spec[0]` contains a file path